src/modules/RecordViewContainer.js

```diff
@@ -0,0 +1,168 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Map } from 'immutable';
+import { NavigationActions } from 'react-navigation';
+import {
+  TouchableOpacity,
+  Image,
+  Text,
+  Dimensions,
+  Alert,
+  View,
+  StyleSheet,
+} from 'react-native';
+import AudioRecorder from '../components/AudioRecorder';
+import LoadingSpinner from '../components/LoadingSpinner';
+import SaveConfirmationWindow from '../components/SaveConfirmationWindow';
+import { addFreeWord } from '../state/UserState';
+import { setText, setAudio } from '../state/HemmoState';
+import { getSessionId } from '../utils/session';
+import { xhr } from '../utils/api';
+import { getSizeByHeight, getImage } from '../services/graphics';
+
+import AppButton from '../components/AppButton';
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    flexDirection: 'column',
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+  audioRecorder: {
+    marginBottom: 70,
+  },
+});
+
+const phrases = require('../data/phrases.json');
+
+const mapStateToProps = state => ({
+  answers: state.getIn(['user', 'currentUser', 'answers']),
+});
+
+const mapDispatchToProps = dispatch => ({
+  back: () => dispatch(NavigationActions.back()),
+  pushRoute: key => dispatch(NavigationActions.navigate({ routeName: key })),
+  popRoute: () => dispatch(NavigationActions.back()),
+  saveFreeWord: freeWord => dispatch(addFreeWord(freeWord)),
+  setText: text => dispatch(setText(text)),
+  setAudio: audio => dispatch(setAudio(audio)),
+});
+
+@connect(mapStateToProps, mapDispatchToProps)
+export default class FreeWordViewContainer extends Component {
+  static navigationOptions = {
+    title: 'Nauhoita',
+  };
+
+  static propTypes = {
+    back: PropTypes.func.isRequired,
+    popRoute: PropTypes.func.isRequired,
+    pushRoute: PropTypes.func.isRequired,
+    setText: PropTypes.func.isRequired,
+    setAudio: PropTypes.func.isRequired,
+    saveFreeWord: PropTypes.func.isRequired,
+    answers: PropTypes.instanceOf(Map).isRequired,
+  };
+
+  state = {
+    text: '',
+    activeWidget: null,
+    showTextForm: false,
+    progress: 0,
+    showSucceedingMessage: false,
+    showSpinner: false,
+  };
+
+  componentWillMount() {
```

Comment:
This componentWillMount can be removed from all three screens as well as the setText & setAudio props